File: Packages/module-ai/Sources/ModuleAI/Domain/Protocols/HistoryRepository.swift
//
//  HistoryRepository.swift
//  ModuleAI
//
//  Created by Codex on 2026/1/1.
//

import Foundation

protocol HistoryRepository {
    func loadHistoryPage(pageNo: Int, pageSize: Int) async throws -> RecordPageModel
    func deleteHistory(historyId: Int) async throws
    func deleteAllHistory() async throws
}

File: app-android/feature/history/src/main/java/com/aidatainsight/android/feature/history/domain/HistoryRepository.kt
package com.aidatainsight.android.feature.history.domain

import com.aidatainsight.android.core.model.contract.RecordPage

interface HistoryRepository {
    suspend fun loadHistoryPage(currentPage: Int, pageSize: Int): RecordPage
    suspend fun deleteHistory(historyId: Int)
    suspend fun deleteAllHistory()
}
